app/controllers/cloud_volume_snapshot_controller.rb

```diff
@@ -0,0 +1,98 @@
+class CloudVolumeSnapshotController < ApplicationController
+  before_action :check_privileges
+  before_action :get_session_data
+  after_action :cleanup_action
+  after_action :set_session_data
+
+  def index
+    redirect_to :action => 'show_list'
+  end
+
+  # handle buttons pressed on the button bar
+  def button
+    @edit = session[:edit] # Restore @edit for adv search box
+    params[:display] = @display if %w(vms instances images).include?(@display)
```

Comment:
this condition doesn't seem right, should be something like volumes, volumes_snapshots?